packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -320,12 +320,19 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
   const needsHorizontalScrollbar = containerWidth && columnsMeta.totalWidth > containerWidth;
 
   const contentSize = React.useMemo(() => {
+    let shouldExtendContent = false;
+    const windowRef = apiRef.current.windowRef;
+    if (windowRef?.current && windowRef.current.scrollHeight <= windowRef.current.clientHeight) {
```

Comment:
I've used `rootRef` instead of `ref`, since `ref` has type `React.Ref<HTMLDivElement>`, which allows it to be a function and I wasn't able to just get `.current` from it.